benchmark-service/benchmark/benchmark.py

```diff
@@ -1,48 +1,126 @@
 from hailtop.config import get_deploy_config
 from hailtop.tls import get_in_cluster_server_ssl_context
 from hailtop.hail_logging import AccessLogger, configure_logging
-from web_common import setup_aiohttp_jinja2, setup_common_static_routes
-
+from web_common import setup_aiohttp_jinja2, setup_common_static_routes, render_template
+from benchmark.utils import ReadGoogleStorage, get_geometric_mean
+import json
+import re
+import plotly
+import plotly.express as px
 
 configure_logging()
 router = web.RouteTableDef()
 logging.basicConfig(level=logging.DEBUG)
 deploy_config = get_deploy_config()
 log = logging.getLogger('benchmark')
 
+BENCHMARK_FILE_REGEX = re.compile(r'gs://hail-benchmarks/((?P<user>[^/]+)/)((?P<version>[^-]+)-)((?P<sha>[^-]+))(-(?P<tag>[^\.]+))?\.json')
+default_filepath = 'gs://hail-benchmarks/tpoterba/0.2.45-ac6815ee857c-master.json'
+
+
+def get_benchmarks(file_path):
+    read_gs = ReadGoogleStorage()
+    try:
+        json_data = read_gs.get_data_as_string(file_path)
+        pre_data = json.loads(json_data)
+    except Exception:
+        message = f'could not find file, {file_path}'
+        log.info('could not get blob: ' + message, exc_info=True)
+        raise web.HTTPBadRequest(text=message)
+
+    data = []
+    prod_of_means = 1
+    for d in pre_data['benchmarks']:
+        stats = dict()
+        stats['name'] = d['name']
+        stats['failed'] = d['failed']
+        if not d['failed']:
+            prod_of_means *= d['mean']
+            stats['f-stat'] = round(d['f-stat'], 6)
+            stats['mean'] = round(d['mean'], 6)
+            stats['median'] = round(d['median'], 6)
+            stats['p-value'] = round(d['p-value'], 6)
+            stats['stdev'] = round(d['stdev'], 6)
+            stats['times'] = d['times']
+            stats['trials'] = d['trials']
+        data.append(stats)
+    geometric_mean = get_geometric_mean(prod_of_means, len(pre_data['benchmarks']))
+
+    file_info = read_gs.parse_file_path(file_path)
+    sha = file_info['sha']
+    benchmarks = dict()
+    benchmarks['sha'] = sha
+    benchmarks['geometric_mean'] = geometric_mean
+    benchmarks['data'] = sorted(data, key=lambda i: i['name'])
+    return benchmarks
+
 
 @router.get('/healthcheck')
 async def healthcheck(request: web.Request) -> web.Response:  # pylint: disable=unused-argument
     return web.Response()
 
 
-@router.get('/{username}')
-async def greet_user(request: web.Request) -> web.Response:
+@router.get('/name/{name}')
+@web_authenticated_developers_only(redirect=False)
+async def show_name(request: web.Request, userdata) -> web.Response:  # pylint: disable=unused-argument
+    query = request.query.get('file')
+    benchmarks = get_benchmarks(query)
+    # benchmarks = get_benchmarks(default_filepath)
+    name_data = next((item for item in benchmarks['data'] if item['name'] == str(request.match_info['name'])),
+                     None)
+
+    def get_x_list():
```

Comment:
Let's move this as a function in utils. It should take an arbitrary list of lists and zip the index of the list. Maybe call it `enumerate_list_index`?